core/include/scipp/core/element/cumulative.h

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Simon Heybrock
+#pragma once
+
+#include "scipp/common/overloaded.h"
+#include "scipp/core/element/arg_list.h"
+#include "scipp/core/transform_common.h"
+
+namespace scipp::core::element {
+
+// Cannot support variances sincetransform copies vale and variance into
+// ValueAndVariance before calling the functor. Only the *first* arg in an
+// in-place transform is copied back to the input.
+constexpr auto exclusive_scan = overloaded{
+    arg_list<double, float, int64_t, int32_t>,
+    transform_flags::expect_no_variance_arg<0>,
+    transform_flags::expect_no_variance_arg<1>, [](auto &sum, auto &x) {
+      sum += x;
+      x = sum - x;
+    }};
+
+constexpr auto inclusive_scan = overloaded{
+    arg_list<int64_t, int32_t>, transform_flags::expect_no_variance_arg<0>,
```

Comment:
Any reason why the `exclusive_scan` can be `double, float` but not the `inclusive_scan`?
If you are exposing `cumsum` to python below, you would expect to be able to do `cumsum` on doubles?

Reply:
Good catch, that is just an oversight.